src/expr/src/scalar/func.rs

```diff
@@ -6991,11 +7013,13 @@ impl From<&VariadicFunc> for ProtoVariadicFunc {
             VariadicFunc::Replace => Replace(()),
             VariadicFunc::JsonbBuildArray => JsonbBuildArray(()),
             VariadicFunc::JsonbBuildObject => JsonbBuildObject(()),
-            VariadicFunc::ArrayCreate { .. } => todo!(),
-            VariadicFunc::ArrayToString { .. } => todo!(),
-            VariadicFunc::ArrayIndex { .. } => todo!(),
-            VariadicFunc::ListCreate { .. } => todo!(),
-            VariadicFunc::RecordCreate { .. } => todo!(),
+            VariadicFunc::ArrayCreate { elem_type } => ArrayCreate(elem_type.into()),
+            VariadicFunc::ArrayToString { elem_type } => ArrayToString(elem_type.into()),
+            VariadicFunc::ArrayIndex { offset } => ArrayIndex(offset.into_proto()),
+            VariadicFunc::ListCreate { elem_type } => ListCreate(elem_type.into()),
+            VariadicFunc::RecordCreate { field_names } => RecordCreate(ProtoRecordCreate {
+                field_names: field_names.iter().map(ProtoColumnName::from).collect(),
```

Comment:
Interesting that you call a `from` here instead of a `Into::into`.

Reply:
I didn't realize that I can do that!